lib/core/configuration.rb

```diff
@@ -17,34 +17,56 @@ class Configuration
     class << self
 
       def update(connector_settings, service_type = nil)
-        if connector_settings.connector_status == Connectors::ConnectorStatus::CREATED
+        if connector_settings.needs_configuration?
```

Comment:
I dont' think this lines up with @wangch079 's [suggestion for how to address this](https://github.com/elastic/connectors-ruby/pull/408#discussion_r1017799108). See also [this slack thread](https://elastic.slack.com/archives/C01795T48LQ/p1669655686117569?thread_ts=1668763646.497289&cid=C01795T48LQ) with @sphilipse  and I.

I'm apprehensive of this change, because it changes the protocol by changing which states are the responsibility of the Connector Service to handle. Similar changes would also need to be made to the Python framework. I don't think we want to do that right now without a decent amount of visibility from from the whole team, and opportunity to comment.